activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -54,16 +54,17 @@ def not(opts, *rest)
       end
     end
 
+    FROZEN_EMPTY_ARRAY = [].freeze
```

Comment:
> Also I don't see any reason to not mutate the values, so I prefer to not do that.

Mutating this value (in the old code) is almost sure to result in incorrect behaviour, since the value is not retained by the receiver. How can you possibly defend this as desired behaviour?

I would like nothing more than to stop bikeshedding on this, but you guys are relentless.